lib/gyp/generator/ninja/index.js

```diff
@@ -573,17 +583,22 @@ NinjaMain.prototype.vars = function vars() {
     cc = 'clang';
     cxx = 'clang++';
   } else if (process.platform === 'win32') {
-    ar = 'lib.exe';
-    ld = 'link.exe';
+    ar = gyp.bindings.which('lib.exe');
```

Comment:
🤔😕
Conceptually you should right, but it failed for me (I have multiple `link.exe` in my path, and ninja used the wrong one), even though `environment.x64` was generated correctly.
If I fix this somehow else, a bunch of stuff can be skipped.